src/test/backend/deploy_test.go

```diff
@@ -177,6 +183,13 @@ func TestDeployAppFromFileWithValidContent(t *testing.T) {
 	if err != nil {
 		t.Errorf("Expected return value to be %#v but got %#v", nil, err)
 	}
+
+	//cleaning the created namespace
+	factory := cmdutil.NewFactory(nil)
```

Comment:
If the test fail for some reason or I stop it in the middle of the work, the namespace will still be there. I think that the only solution is to create an interface for the dashboard <-> kubectl boundary.  For example, you could create an interface for the line:
`_, err = kubectlResource.NewHelper(info.Client, info.Mapping).Create(info.Namespace, true, info.Object)`

In tests you'd pass a verifying callback. On prod you'd call kubectl implementation callback.